web/shared/src/fenced_code.ts

```diff
@@ -210,14 +230,15 @@ export function process_fenced_code(content) {
 
     for (const line of input) {
         const handler = handler_stack.at(-1);
+        assert(handler !== undefined, "Handler stack is empty.");
         handler.handle_line(line);
     }
 
     // Clean up all trailing blocks by letting them
     // insert closing fences
     while (handler_stack.length !== 0) {
         const handler = handler_stack.at(-1);
-        handler.done();
+        handler!.done();
```

Comment:
Why is this one doing this, vs. the `assert` statement above?

Reply:
It can be seen that the `handler` inside the while loop will not be `null` because we are only accessing the last element of the handler_stack when it has at least one element in it. Hence, here I have used `!` instead of the `assert`.